### PR TITLE
Fix tabs ID handling in scripting engine

### LINKED ISSUE #4268: Dialog Tabs lose reference or ID after changing their text with `dialog:modify`
It seems that a reference to tabs contets is getting lost (or tab ID is getting set to null) after chaning a tab text with `dialog:modify`.

Sample code:
```lua
function TabsDialog(tabs)
    local dialog = Dialog("Tabs")

    for i, tab in ipairs(tabs) do
        dialog --
        :tab{id = tostring(i), text = tab.text, onclick = function() end} --
        :entry{
            id = "name-" .. tostring(i),
            label = "Name",
            text = tab.text,
            focus = true,
            onchange = function()
                local name = dialog.data["name-" .. tostring(i)]
                tab.text = name
                dialog:modify{id = tostring(i), text = name}

                print("Tab ID = ", i, "Text = ", name)
            end
        } --
        :button{
            text = "Delete",
            onclick = function()
                table.remove(tabs, i)

                dialog:close()
                local dialog = TabsDialog(tabs)
                dialog:show{wait = false}
            end
        }
    end

    dialog:tab{
        id = "add-tab",
        text = "+",
        onclick = function()
            table.insert(tabs, {text = "New"})
            dialog:close()
            local dialog = TabsDialog(tabs)
            dialog:show{wait = false}
        end
    }

    dialog:endtabs{
        id = "tabs",
        selected = "tab",
        align = Align.LEFT,
        onchange = function(ev) print("Selected tab ID = ", ev.tab) end
    }

    return dialog
end

local tabs = {{text = "Tab #1"}, {text = "Tab #2"}}

local dialog = TabsDialog(tabs)
dialog:show{wait = false}

print("API Version = ", app.apiVersion)
```
Preview:
tabs-reference-bug
tabs-reference-bug2

### Aseprite and System version

* Aseprite version: v1.3.2, Steam Beta
* System: Windows 11

### PR DESCRIPTION
**Pull request by martincapello**
Fix #4268

### CODE REVIEW

**Review by Gasparoken — APPROVED** [review]
LGTM.

File: src/app/script/tabs_widget.cpp [hunk 1]
```diff
@@ -31,18 +34,22 @@ Tabs::Tabs(int selectorFlags) : m_selectorFlags(selectorFlags)
   layoutChilds();
 }
 
-Tab* Tabs::addTab(Grid* content)
+Tab* Tabs::addTab(std::string id, std::string text)
```
Comment (dacap) [inline src/app/script/tabs_widget.cpp:37]:
```suggestion
Tab* Tabs::addTab(const std::string& id, const std::string& text)
```

File: src/app/script/tabs_widget.h [hunk 2]
```diff
@@ -37,7 +44,7 @@ class Tabs : public ui::VBox {
 
   Tabs(int selectorFlags);
 
-  Tab* addTab(ui::Grid* content);
+  Tab* addTab(std::string id, std::string text);
```
Comment (dacap) [inline src/app/script/tabs_widget.h:47]:
```suggestion
  Tab* addTab(const std::string& id, const std::string& text);
```

File: src/app/script/tabs_widget.cpp [hunk 3]
```diff
@@ -31,18 +34,22 @@ Tabs::Tabs(int selectorFlags) : m_selectorFlags(selectorFlags)
...
     m_buttonsBox.addChild(&m_buttons);
-    m_buttons.ItemChange.connect([this](ButtonSet::Item* selItem) {
+    m_buttons.ItemChange.connect([this](ButtonSet::Item* tab) {
       int oldSelectedTabIndex = m_selectedTab;
       for (int i=0; i<m_pages.children().size(); ++i) {
-        auto tab = m_pages.children()[i];
-        bool isSelectedTab = selItem->text() ==  tab->text();
-        tab->setVisible(isSelectedTab);
+        auto tabContent = m_pages.children()[i];
+        bool isSelectedTab = TAB_CONTENT_ID(tab->id()) == tabContent->id();
```
Comment (dacap) [inline src/app/script/tabs_widget.cpp:51]:
```suggestion
        bool isSelectedTab = (TAB_CONTENT_ID(tab->id()) == tabContent->id());
```

**Review by dacap — CHANGES_REQUESTED** [review]
Minor changes.

**Pull request merged**